doit/dependency.py

```diff
@@ -361,6 +338,96 @@ def remove_all(self):
         self._conn.execute('delete from doit')
 
 
+class BaseChecker(object):
+    """Base Checker that must be inherited."""
+
+    def __init__(self, backend):
```

Comment:
I think it is not necessary to take the `backend` as a parameter.

on `save_state` pass the current state, and return the state to be saved...
`save_state()` was probably not a good name.